pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/QueryRunner.java

```diff
@@ -94,12 +98,19 @@ public void processQuery(DistributedStagePlan distributedStagePlan, ExecutorServ
           ServerRequestUtils.constructServerQueryRequest(distributedStagePlan, requestMetadataMap);
 
       // send the data table via mailbox in one-off fashion (e.g. no block-level split, one data table/partition key)
-      DataTable dataTable = _serverExecutor.processQuery(serverQueryRequest, executorService, null);
+      BaseDataBlock dataBlock;
+      try {
+        DataTable dataTable = _serverExecutor.processQuery(serverQueryRequest, executorService, null);
+        // this works because default DataTableImplV3 will have ordinal 0, which maps to ROW(0)
+        dataBlock = DataBlockUtils.getDataBlock(ByteBuffer.wrap(dataTable.toBytes()));
+      } catch (IOException e) {
+        throw new RuntimeException("Unable to convert byte buffer", e);
```

Comment:
(nit) suggest using `Unable to convert byte buffer to transferable block in the leaf stage`. I think we should ideally include some metadata about the stage in exception or do error level log for that -- stageId, endpoint name etc